satellite/satellitedb/migrate.go

```diff
@@ -0,0 +1,330 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"database/sql"
+	"strconv"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/migrate"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/satellite/console"
+)
+
+// ErrMigrate is for tracking migration errors
+var ErrMigrate = errs.Class("migrate")
+
+// CreateTables is a method for creating all tables for database
+func (db *DB) CreateTables() error {
+	switch db.driver {
+	case "postgres":
+		migration := db.PostgresMigration()
+		return migration.Run(db.log.Named("migrate"), db.db)
+	default:
+		return migrate.Create("database", db.db)
+	}
+}
+
+// PostgresMigration returns steps needed for migrating postgres database.
+func (db *DB) PostgresMigration() *migrate.Migration {
```

Comment:
Is this using something from satellitedb.DB interface? Maybe can be just `satellitedb.PostgresMigration()`?

Reply:
It's intentionally not exposed on the database interface. `satellite.Peer` shouldn't care how the database handles migrations internally.